experimental/single_number_formatter/src/ule.rs

```diff
@@ -7,10 +7,11 @@ use zerovec::{
     ule::{AsULE, ZeroVecError, ULE},
 };
 
-use crate::provider::{
-    CurrencyPatterns, PatternSelection, PlaceHolder, MAX_PLACE_HOLDER_INDEX, NO_PLACE_HOLDER,
-    USE_ISO_CODE,
-};
+use crate::provider::{CurrencyPatterns, PatternSelection, PlaceHolder};
+
+const NO_PLACE_HOLDER: u16 = 0b0111_1111_1111;
+const USE_ISO_CODE: u16 = 0b0111_1111_1110;
+pub const MAX_PLACE_HOLDER_INDEX: u16 = 0b0111_1111_1101;
```

Comment:
I guess this constant should stay with `PlaceHolder` now

Reply:
The thing you're calling `PlaceHolder` should have an internal invariant that the index is with range and a fallible constructor that checks the invariant. Then you can get rid of the `panic!` in the `to_unaligned` function as well as the `assert!` in datagen.

Reply:
It's a bit annoying because it's an enum and we won't be able to match on it if the variants are private, but I guess that is the cleanest solution.

Reply:
If you want to keep the public facing type an enum, you could do

```rust
enum SymbolPlace {
    Iso,
    Index(SymbolIndex),
}

struct SymbolIndex(u16);

impl TryFrom<u16> for SymbolIndex {
    type Error = CurrencyError;
    fn try_from(input: u16) -> Self {
        if input > MAX_PLACE_HOLDER_INDEX {
            Err(CurrencyError::Limit)
        } else {
            Ok(Self(input))
        }
    }
}
```

Reply:
Another thing you can do with this model is make the `u16` a nichey type like `NonZeroU16` to reduce the size of `SymbolPlace`

Reply:
Okay, this work I will defer it to the next PR. 

here is the issue: 
https://github.com/unicode-org/icu4x/issues/4013